app/frontend/src/features/profile/view/ProfilePage.test.tsx

```diff
@@ -0,0 +1,169 @@
+  within,
+} from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { Empty } from "google-protobuf/google/protobuf/empty_pb";
+import { Route } from "react-router-dom";
+import { profileRoute } from "routes";
+import { service } from "service/index";
+import { getHookWrapperWithClient } from "test/hookWrapper";
+import { getUser } from "test/serviceMockDefaults";
+import { addDefaultUser, MockedService } from "test/utils";
+
+import {
+  CANCEL,
+  MORE_PROFILE_ACTIONS_A11Y_TEXT,
+  REPORT_DETAILS,
+  REPORT_REASON,
+  REPORT_USER,
+  SEND,
+} from "../constants";
+import ProfilePage from "./ProfilePage";
+
+const getUserMock = service.user.getUser as MockedService<
+  typeof service.user.getUser
+>;
+const reportUserMock = service.user.reportUser as MockedService<
+  typeof service.user.reportUser
+>;
+
+function renderProfilePage(username?: string) {
+  const { wrapper } = getHookWrapperWithClient({
+    initialRouterEntries: [`${profileRoute}${username ? `/${username}` : ""}`],
+  });
+
+  render(
+    <Route path={`${profileRoute}/:username?`}>
+      <ProfilePage />
+    </Route>,
+    { wrapper }
+  );
+}
+
+describe("Profile page", () => {
+  beforeAll(() => {
+    jest.setTimeout(10000);
+  });
+
+  beforeEach(() => {
+    getUserMock.mockImplementation(getUser);
+    reportUserMock.mockResolvedValue(new Empty());
+    addDefaultUser();
+  });
+
+  describe("when viewing the current user's profile", () => {
+    it("does not show the button for opening a profile actions menu", async () => {
+      renderProfilePage();
+
+      expect(
+        await screen.findByRole("heading", { name: "Funny Cat current User" })
+      ).toBeVisible();
+      expect(
+        screen.queryByRole("button", { name: MORE_PROFILE_ACTIONS_A11Y_TEXT })
+      ).not.toBeInTheDocument();
+    });
+
+    it("does not show the button for opening a profile actions menu (viewing via current user's username)", async () => {
+      renderProfilePage("funnycat");
+
+      expect(
+        await screen.findByRole("heading", { name: "Funny Cat current User" })
+      ).toBeVisible();
+      expect(
+        screen.queryByRole("button", { name: MORE_PROFILE_ACTIONS_A11Y_TEXT })
+      ).not.toBeInTheDocument();
+    });
+  });
+
+  describe("when viewing another user's profile", () => {
+    beforeEach(() => {
+      renderProfilePage("funnydog");
+    });
+
+    it("shows the button for opening a profile actions menu if viewing another user's profile", async () => {
+      expect(
+        await screen.findByRole("heading", { name: "Funny Dog" })
+      ).toBeVisible();
+      expect(
+        screen.getByRole("button", { name: MORE_PROFILE_ACTIONS_A11Y_TEXT })
+      ).toBeVisible();
+    });
+
+    it("opens the profile actions menu when the linked button is clicked", async () => {
+      userEvent.click(
+        await screen.findByRole("button", {
+          name: MORE_PROFILE_ACTIONS_A11Y_TEXT,
+        })
+      );
+
```

Comment:
nit: delete extra line?

Reply:
I think I like to keep the extra line for this one - it's an implicit break for the arrange-act-assert test structure but without writing `// act` and `// assert` comments everywhere.